tools/SourceKit/docs/Protocol.md

```diff
@@ -54,6 +54,10 @@ provided.
 }
 ```
 
+| Request Name | Request Key | Description |
+| -------------:|:------------|:------------|
+| `open` | `complete.open` | Given an input file, will determine at each offset what keyword is possible. Default offset is `0` if none is provided. |
```

Comment:
I suggest we move this table up above the codecomplete request description, and add both the original `codecomplete` request and the newer `codecomplete.open`, `codecomplete.close`, and `codecomplete.update` requests.  If you only want to cover `codecomplete.open` in this PR, that's fine, but we should at least add `codecomplete` as well.

Please note, the request key is `codecomplete.open`, not just `complete.open`.  For the description, I suggest something like "Open a code-completion session for the given input file and offset, and return the initial list of completions".  The key difference between `codecomplete.open` and `codecomplete` is that the former opens a session that can filter completions using `codecomplete.update` and must be closed with `codecomplete.close`, whereas the latter just returns all of the results without filtering.

We should also describe what else goes in the request, similar to the description of the `codecomplete` request, or at least describe the differences between that request and this one (I think the only difference in the request structure is that `codecomplete.open` can take an options dictionary).